issue_drafts/issue_4.md

```diff
@@ -0,0 +1,126 @@
+# Issue 4: Generalizing Rules with Wildcards
+
+Pipeline steps: Compute the mean predicted temperature on each day of the year (at all depths) for each lake of interest
+
+Concepts learned:
+- wildcards
+- rule all
+- dry-run
+
+## Update `calc_doy_means.py` script to use properties from Snakemake rule
+Our current workflow is able to fetch and unzip raw lake temperature prediction data from ScienceBase. In the `out` file of our `1_fetch` step, we have a folder called `tmp` containing a set of csv files, each of which contains daily temperature predictions for a single lake (at multiple depths).
+
+The daily data spans nearly 40 years, and we would like to process it by calculating the day-of-year mean temperature predictions (at all depths) for each lake. Let's add a new rule to do this.
+
+For this tutorial, we will choose two lakes that we want to calculate the day-of-year mean temperatures for. The lakes that we want to process have the IDs `120020150` and `107072210`.
+
+Let's process the data based on our current knowledge of Snakemake pipelining. We can write a rule that uses the `2_process/calc_doy_means.py` script. We will need to indicate the input and output files for the two lakes we have chosen in our rule as well:
+```
+rule calc_doy_means:
+    input:
+        "1_fetch/out/tmp/pgdl_nhdhr_120020150_temperatures.csv",
+        "1_fetch/out/tmp/pgdl_nhdhr_107072210_temperatures.csv"
+    output:
+        "2_process/out/doy_120020150.csv",
+        "2_process/out/doy_107072210.csv"
+    script:
+        "2_process/calc_doy_means.py"
+```
+
+This rule is perfectly acceptable and should work if you test it out by running `snakemake --cores 1 calc_doy_means` in your command line/Anaconda prompt.
+
+Note: If you do test out the rule, be sure to delete the outputs before moving forward (`snakemake --cores 1 calc_doy_means --delete-all-output`). If you don't delete the outputs, the Snakemake pipeline will not execute next time because the outputs have already been generated!
```

Comment:
```suggestion
> Note: If you do test out the rule, be sure to delete the outputs before moving forward (`snakemake --cores 1 calc_doy_means --delete-all-output`). If you don't delete the outputs, the Snakemake pipeline will not execute next time because the outputs have already been generated!
```
For style consistency